### PR TITLE
Fix to Issue 1690

### PR DESCRIPTION
**Pull request by hesingh**
The PR includes

(a)  Changes to p4c/control-plane/CMakeLists.txt to include .cpp files.
(b) Changes to some .cpp files in p4c/control-plane/ to pass cpplint now that cpplint is enabled on the files.

### CODE REVIEW

**Comment by hesingh** [comment]
Travis has passed.  Please review - thanks.

File: control-plane/CMakeLists.txt [hunk 1]
```diff
@@ -83,19 +83,25 @@ set (CONTROLPLANE_SOURCES
   ${CONTROLPLANE_SRCS}
   )
 
-set (CONTROLPLANE_HDRS
+set (CONTROLPLANE_SRCS
```
Comment (antoninbas) [inline control-plane/CMakeLists.txt]: I don't understand this change. `CONTROLPLANE_SRCS` is defined above already.

Comment (hesingh) [reply control-plane/CMakeLists.txt]: cmake is running top to down in the file.  There is no further use of `CONTROLPLANE_SRCS` except for cpplint, so it should be fine.

However, it confuses code.  I  will change code.

**Comment by hesingh** [comment]
> This goes against our Cmake naming convention. How to add `${CONTROLPLANE_HDRS}` to add_cpplint_files

Please see a new checkin in which I have restored naming convetion.  I hope the new changes work for you.

**Comment by hesingh** [comment]
I believe I have addressed all comments.  Are there any other?  If not, please approve.

**Comment by hesingh** [comment]
@antoninbas Any concerns with this PR?  Please try and close the review - thanks.

**Comment by hesingh** [comment]
@cc10512 thanks for the review.  I don't think anyone else has any other comments.

Please checkin this code when you get a chance - thanks.

**Comment by hesingh** [comment]
@jafingerhut I think this PR has been beaten into submission.   Please merge when you get a chance - thanks.

**Comment by jafingerhut** [comment]
@hesingh I am loathe to merge in changes that I haven't reviewed myself, and am personally confident of.  I would much rather someone who knows a non-0 amount about the changes do so (for this particular PR, I am also not very motivated to learn enough about the p4c build system to be personally confident of the changes made here).

**Comment by hesingh** [comment]
Ok, @jafingerhut No worries - sorry to bother you.  I asked @cc10512 in the morning, but no word yet from him and this is why I thought I'd ask another person.

**Comment by hesingh** [comment]
@hanw or @antoninbas , if you are ok with the changes, please approve.  Then either one of you or @cc10512 can merge this PR.  thanks.

**Comment by cc10512** [comment]
@hesingh this is not such a critical fix that I need to override @antoninbas's request for changes. You need to get him to approve and we'll merge it then.

**Pull request merged**